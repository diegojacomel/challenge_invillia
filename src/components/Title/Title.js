/** Modules */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const HeaderStyle = styled.h1`
    display: block;
    line-height: 1.2;
    font-family: 'Open Sans', Arial, sans-serif;
    color: ${props => props.theme.colors[props.color]};
    text-align: ${props => props.alignCenter ? "center" : "left"};
    font-size: ${props => props.theme.fontSize[props.fontSize]};
    margin: ${props => props.marginTop ? `${props.theme.spacing.md} ${props.theme.spacing.none} ${props.theme.spacing.none}` : "0"};
`

const Header = ({ children, color, alignCenter, fontSize, marginTop }) => (
    <HeaderStyle color={color} alignCenter={alignCenter} fontSize={fontSize} marginTop={marginTop}>
        {children}
    </HeaderStyle>
)

function Title({ children, color, alignCenter, fontSize, marginTop }) {
    return (
        <Header color={color} alignCenter={alignCenter} fontSize={fontSize} marginTop={marginTop} id="testTitle">
            {children}
        </Header>
    )
}

Title.propTypes = {
    color: PropTypes.string,
    alignCenter: PropTypes.bool,
    fontSize: PropTypes.string,
    marginTop: PropTypes.bool
}

Title.defaultProps = {
    color: "primary",
    alignCenter: false,
    fontSize: "lg",
    marginTop: false
}

export default Title;
