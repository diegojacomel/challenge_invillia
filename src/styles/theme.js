const primaryTheme = {
    colors: {
        primary: '#007bff',
        success: '#28a745',
        secondary: '#6c757d',
        warning: "#ffc107",
        black: '#000000',
        white: '#FFFFFF',
        gray: '#cccccc',
        grayLight: '#F2F2F2',
        grayDark: '#333333'
    },
    fontSize: {
        xs: "14px",
        sm: "16px",
        md: "20px",
        lg: "25px",
        hg: "40px",
        hhg: "60px"
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '80px',
        ggt: '100px',
        hg: '200px',
        hhg: '400px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    },
    doubleSpacing: {
        xs: "2px 5px",
        sm: "3px 10px",
        md: "5px 12px",
        lg: "8px 16px",
        xg: "10px 20px",
        hg: "20px 40px"
    },
    rounded: {
        none: '0',
        xs: '2px',
        sm: '3px',
        md: '5px',
        lg: '10px'
    }
}

export {
    primaryTheme
}