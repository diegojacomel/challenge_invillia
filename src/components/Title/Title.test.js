// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/theme';

// Component
import Title from './Title';

describe('Title', () => {
    const wrapper = shallow(<Title />);

    test('Render without crashing', () => {
        shallow(<Title />);
    })

    test('It exist', () => {
        expect(wrapper.find('#testTitle')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('#testTitle');
    })

    test('Have display name', () => {
        expect(wrapper.find('#testTitle')).toHaveDisplayName('Header');
    })

    test('Match selector', () => {
        expect(wrapper.find('#testTitle')).toMatchSelector('Header');
    })

    test('Have props', () => {
        expect(wrapper.find('#testTitle')).toHaveProp('color');
        expect(wrapper.find('#testTitle')).toHaveProp('alignCenter');
        expect(wrapper.find('#testTitle')).toHaveProp('fontSize');
        expect(wrapper.find('#testTitle')).toHaveProp('marginTop');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Title color="black" alignCenter={true} fontSize="lg">
                    Title Example
                </Title>
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})