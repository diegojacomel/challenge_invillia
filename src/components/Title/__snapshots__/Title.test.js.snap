// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Render correctly 1`] = `
<h1
  className="sc-bdVaJa khxkCA"
  color="black"
  fontSize="lg"
>
  Title Example
</h1>
`;
